packages/iris-grid/src/sidebar/visibility-ordering-builder/sortable-tree/SortableTree.tsx

```diff
@@ -0,0 +1,350 @@
+                indicator={indicator}
+                collapsed={Boolean(collapsed && children.length)}
+                onCollapse={
+                  collapsible && children.length
+                    ? () => handleCollapse(id)
+                    : undefined
+                }
+                onRemove={
+                  removable && onRemove ? () => handleRemove(id) : undefined
+                }
+                item={item}
+                renderItem={renderItem}
+              />
+            );
+          })}
+        </div>
+        {createPortal(
+          <DragOverlay
+            dropAnimation={dropAnimationConfig}
+            modifiers={indicator ? [adjustTranslate] : undefined}
+          >
+            {activeId && activeItem ? (
+              <SortableTreeItem
+                id={activeId}
+                depth={activeItem.depth}
+                clone
+                childCount={getChildCount(items, activeId) + 1}
+                value={activeId.toString()}
+                indentationWidth={indentationWidth}
+                renderItem={renderItem}
+                item={activeItem}
+              />
+            ) : null}
+          </DragOverlay>,
+          document.body
+        )}
+      </SortableContext>
+    </DndContext>
+  );
+
+  function handleDragStart({ active: { id: newActiveId } }: DragStartEvent) {
+    setActiveId(newActiveId);
+    setOverId(newActiveId);
+
+    document.body.style.setProperty('cursor', 'grabbing');
+  }
+
+  function handleDragMove({ delta }: DragMoveEvent) {
+    setOffsetLeft(delta.x);
+  }
+
+  function handleDragOver({ over }: DragOverEvent) {
+    setOverId(over?.id ?? null);
+  }
+
+  function handleDragEnd({ active, over }: DragEndEvent) {
+    resetState();
+
+    if (projected && over) {
+      const { depth, parentId } = projected;
+
+      const clonedItems: FlattenedItem<T>[] = JSON.parse(
+        JSON.stringify(flattenTree(items))
+      );
+      const overIndex = clonedItems.findIndex(({ id }) => id === over.id);
+      const activeIndex = clonedItems.findIndex(({ id }) => id === active.id);
+      const activeTreeItem = clonedItems[activeIndex];
+
+      clonedItems[activeIndex] = { ...activeTreeItem, depth, parentId };
+
+      onChange?.(activeTreeItem, {
+        ...clonedItems[overIndex],
+        parentId: projected.parentId,
+      });
+      const sortedItems = arrayMove(clonedItems, activeIndex, overIndex);
+      const newItems = buildTree(sortedItems);
+
+      setItems(newItems);
+    }
+  }
+
+  function handleDragCancel() {
+    resetState();
+  }
+
+  function resetState() {
+    setOverId(null);
+    setActiveId(null);
+    setOffsetLeft(0);
+
+    document.body.style.setProperty('cursor', '');
+  }
+
+  function handleCollapse(id: UniqueIdentifier) {
+    setItems(oldItems =>
+      setProperty(oldItems, id, 'collapsed', value => !value)
+    );
+  }
+}
+
+const adjustTranslate: Modifier = ({ transform }) => {
```

Comment:
I'd prefer `function adjustTranslate`

Reply:
It's typed this way because `Modifier` is a function type. Changing would have to be `function adjustTranslate({ transform }: Parameters<Modifier>[0]): ReturnType<Modifier> { ... }` which is much more fragile